taichi/program/ndarray.cpp

```diff
@@ -147,9 +147,9 @@ T Ndarray::read(const std::vector<int> &I) const {
       staging_buf_->get_ptr(),
       this->ndarray_alloc_.get_ptr(/*offset=*/index * sizeof(T)), size_);
 
-  char *const device_arr_ptr =
-      reinterpret_cast<char *>(staging_buf_->device->map(*staging_buf_));
-  TI_ASSERT(device_arr_ptr);
+  char *device_arr_ptr;
```

Comment:
nit: can we initialize all these local pointers? We plan to enable `https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines/init-variables.html` in clang-tidy check in the future.